Resources/public/js/views/fields/ez-relationlist-editview.js

```diff
@@ -132,17 +132,33 @@ YUI.add('ez-relationlist-editview', function (Y) {
                 relatedContentsJSON = [];
 
             Y.Array.each(relatedContents, function (value) {
-                relatedContentsJSON.push(value.toJSON());
-            });
+                var relatedContentJSON = value.toJSON();
+
+                if (this._isNewRelation(value) || relatedContentJSON.resources.MainLocation) {
+                    relatedContentsJSON.push(relatedContentJSON);
+                }
+            }, this);
 
             return {
-                relatedContents:  relatedContentsJSON,
+                relatedContents: relatedContentsJSON,
                 loadingError: this.get('loadingError'),
                 isEmpty: this._isFieldEmpty(),
                 isRequired: this.get('fieldDefinition').isRequired,
             };
         },
 
+        /**
+         * Check if relation to the content is new (non-existed before edit)
+         *
+         * @method _isNewRelation
+         * @protected
+         * @param {Object} object
```

Comment:
What object? The param description is meaningless in this case. Please, name the param correctly and/or give it a meaningful description.